### PR TITLE
Update mf/react to work with shared assets

### PR DESCRIPTION
**Pull request by pmallett-mc**
Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)

## Description
Updates mf/react to use the shared assets in the project root.
During builds, assets are copied into src/assets so that they can be bundled in /es/assets, then they are cleaned up so that they only exist at /assets to avoid confusion.

The UMD build (unused) is removed, it was not working with path aliases

When `npm start` is run, a symbolic link is created instead because cleanup cannot be done reliably. This also obviates the need for a special assets-watch update during local dev.

## Related Issue / Ticket

- [JIRA issue]()
- [Github issue]()

## Steps to Test

1. check out code, run all `npm run` commands
1. everything should work as expected:
** npm run build should result in a /es/ build where paths resolve to /es/assets
** npm start should work because of the /src/assets symbolic link
** SharedAssets alias will work during npm start and storybook builds because of the resolver in react/.storybook/webpack.config.js 
** npm lint will work because of the path alias resolver in react/.eslintrc.json
** in case src/assets is accidentally left in place during dev, it is git ignored so duplicated assets will not be committed
** SharedAssets alias is handled during npm build by the setup in react/nwb.config.js

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application

* 

#### @TODO

*

### CODE REVIEW

File: react/package.json [hunk 1]
```diff
@@ -59,13 +60,17 @@
     "acorn-object-rest-spread": "^1.1.0",
     "acorn-static-class-property-initializer": "^1.0.0",
     "babel-core": "^6.26.0",
+    "babel-plugin-module-resolver": "^3.1.1",
+    "babel-plugin-transform-import-paths": "^1.0.0",
     "bourbon": "^5.1.0",
```
Comment (clairesunstudio) [inline react/package.json]: looks good! The only thing I would say is move `bourbon` `bourbon-neat` and `normalize-scss` into build dependencies, so when we compile css in search or EEC we don't need to install them separately

File: react/scripts/assets-copy.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const fse = require('fs-extra');
+
+fse.remove('src/assets', (err1) => {
```
Comment (clairesunstudio) [inline react/scripts/assets-copy.js]: and based on our convo sounds like we don't need the remove step here

**Review by smurrayatwork — APPROVED** [review]

**Comment by clairesunstudio** [comment]
screen shot 2018-09-19 at 10 24 15 am
still getting this error ^ we can look into this later

**Comment by clairesunstudio** [comment]
Symlink method with `--copy-files` is probably not going to work for windows system. We can verify that and follow up when necessary

**Review by clairesunstudio — APPROVED** [review]
Otherwise looks great! exciting stuff! thanks @petermallett

**Pull request merged**